engines/scumm/palette.cpp

```diff
@@ -1381,6 +1381,76 @@ const byte *ScummEngine::getPalettePtr(int palindex, int room) {
 	return cptr;
 }
 
+uint32 ScummEngine::getPaletteColorFromRGB(byte *palette, byte r, byte g, byte b) {
+	uint32 color, black = 0x00, white = 0xFF;
```

Comment:
Thanks, in the following days I'll take a look at this and build a system like this!

Reply:
@AndywinXp coincidentally, I need exactly this functionality for Director, so I will implement this class today, so then you can just reuse it.

Reply:
Done, it now lives in `graphics/palette.h`

Reply:
Thanks sev!

Reply:
I have been experimenting with `PaletteLookup`: it works great for most part, but I have found that the best-color algorithm found there sometimes returns different values than the one produced by the SCUMM more-naive version of the same algorithm (the one that we're reviewing right now).

This could be solved in different ways:
1) Add the possibility to use this simpler algorithm in PaletteLookup;
2) Building a SCUMM-only version of the lookup methods;
3) Avoid using the palette lookup for now, since it would only be used one time for each time a banner is displayed (in fact it isn't used anywhere else);

What do you suggest?